packages/jss-plugin-nested/src/index.js

```diff
@@ -98,13 +98,16 @@ export default function jssNested(): Plugin {
         container.addRule(selector, style[prop], {...options, selector})
       } else if (isNestedConditional) {
         // Place conditional right after the parent rule to ensure right ordering.
-        container
-          .addRule(prop, {}, options)
-          // Flow expects more options but they aren't required
-          // And flow doesn't know this will always be a StyleRule which has the addRule method
-          // $FlowFixMe[incompatible-use]
-          // $FlowFixMe[prop-missing]
-          .addRule(styleRule.key, style[prop], {selector: styleRule.selector})
+        const queue = container.prepareQueue()
+        const addedRule = container.createRule(prop, {}, options)
+
+        // Flow expects more options but they aren't required
+        // And flow doesn't know this will always be a StyleRule which has the addRule method
+        // $FlowFixMe[incompatible-use]
+        // $FlowFixMe[prop-missing]
+        addedRule.addRule(styleRule.key, style[prop], {selector: styleRule.selector})
```

Comment:
Essentially all this is about ability to call an .addRule on a rule that implements ContainerRule without rendering anything (only adding to a queue) and then deploying the container rule as a next step. Is that correct?

I think it would be much easier to find a better API that allows this if we had tests to verify the behavior, because  right now trying things out without a test is way too hard.